structuralcodes/geometry/_geometry.py

```diff
@@ -208,10 +228,43 @@ def rotate(
             ),
             diameter=self._diameter,
             material=self._material,
+            density=self._density,
             name=self._name,
             group_label=self._group_label,
         )
 
+    @staticmethod
+    def from_geometry(
+        geo: PointGeometry,
+        new_material: t.Optional[t.Union[Material, ConstitutiveLaw]] = None,
+    ) -> PointGeometry:
+        """Create a new PointGeometry with a different material."""
+        if not isinstance(geo, PointGeometry):
+            raise TypeError('geo should be a PointGeometry')
+        if new_material is not None:
+            # provided a new_material
+            if not isinstance(new_material, Material) and not isinstance(
+                new_material, ConstitutiveLaw
+            ):
+                raise TypeError(
+                    f'new_material should be a valid structuralcodes.base.\
+                    Material or structuralcodes.base.ConstitutiveLaw object. \
+                    {repr(new_material)}'
+                )
+        else:
+            # new_material not provided, assume elastic material with same
+            # elastic modulus
+            new_material = Elastic(E=geo.material.get_tangent(eps=0)[0])
```

Comment:
This behaviour of creating a linear elastic material with the initial tangent as stiffness should be documented in the docstring.